stellar/its.js

```diff
@@ -183,23 +185,124 @@ async function execute(wallet, _, chain, contract, args, options) {
     await broadcast(operation, wallet, chain, 'Executed', options);
 }
 
+// FIXME: Remove this after v1.1.1 release
 async function migrateTokens(wallet, _, chain, contract, args, options) {
     let tokenIds = Array.isArray(args) ? args : [args];
 
     tokenIds = tokenIds.map((tokenId) => '0x'.concat(Buffer.from(tokenId, 'base64').toString('hex')));
 
     for (const tokenId of tokenIds) {
         printInfo('Migrating token', tokenId);
+        printInfo('Upgrader address', chain.contracts.Upgrader.address);
 
-        const tokenIdScVal = nativeToScVal(Buffer.from(tokenId, 'hex'));
+        const tokenIdScVal = hexToScVal(tokenId);
         const upgraderAddressScVal = nativeToScVal(Address.fromString(chain.contracts.Upgrader.address), { type: 'address' });
         const newVersionScVal = nativeToScVal(options.version, { type: 'string' });
 
-        const operation = contract.call('migrate_token', tokenIdScVal, upgraderAddressScVal, newVersionScVal);
+        const tokenManagerAddressOperation = contract.call('token_manager_address', tokenIdScVal);
+        const tokenManagerAddressResult = await broadcast(
+            tokenManagerAddressOperation,
+            wallet,
+            chain,
+            'Retrieved TokenManager address',
+            options,
+        );
+        const tokenManagerAddress = serializeValue(tokenManagerAddressResult.value());
+
+        printInfo('TokenManager address', tokenManagerAddress);
+
+        const interchainTokenAddressOperation = contract.call('interchain_token_address', tokenIdScVal);
+        const interchainTokenAddressResult = await broadcast(
+            interchainTokenAddressOperation,
+            wallet,
+            chain,
+            'Retrieved InterchainToken address',
+            options,
+        );
+        const interchainTokenAddress = serializeValue(interchainTokenAddressResult.value());
+
+        printInfo('InterchainToken address', interchainTokenAddress);
+
+        const auths = await createMigrateTokenAuths(
+            contract,
+            tokenManagerAddress,
+            interchainTokenAddress,
+            tokenIdScVal,
+            upgraderAddressScVal,
+            newVersionScVal,
+            chain,
+            wallet,
+        );
+
+        const operation = Operation.invokeContractFunction({
+            contract: chain.contracts.InterchainTokenService.address,
+            function: 'migrate_token',
+            args: [tokenIdScVal, upgraderAddressScVal, newVersionScVal],
+            auth: auths,
+        });
+
         await broadcast(operation, wallet, chain, 'Migrated token', options);
     }
 }
 
+// FIXME: Remove this after v1.1.1 release
```

Comment:
```suggestion
// TODO: Remove this after v1.1.1 release
```
easier to grep one type

Reply:
updated